openstack/utils/choose_version.go

```diff
@@ -109,3 +113,94 @@ func ChooseVersion(client *gophercloud.ProviderClient, recognized []*Version) (*
 
 	return highest, endpoint, nil
 }
+
+// GetSupportedMicroversions returns the minimum and maximum microversion that is supported by the ServiceClient Endpoint.
+func GetSupportedMicroversions(client gophercloud.ServiceClient) (string, string, error) {
```

Comment:
(Thinking aloud) maybe we could return a wrapper structure with something like
```go
type SupportedMicroversions struct {
    MaxMajor int
    MaxMinor int
    MinMajor int
    MinMinor int
}
```

Then some of these functions become methods on it, like
```go
func (vs SupportedMicroversions) Require(version string) error
func (vs SupportedMicroversions) IsSupported(version string) bool
func (vs SupportedMicroversions) Choose(versions []string) string
```

etc

Reply:
Also: you want to pass the client as a pointer, not by value.

Reply:
Would the idea be to hold on to this wrapper and pass it around? I'm worried it would become hard to use. If I need to check/require microversions in many different places in a code base it would probably be cumbersome. Alternatively, I would end up calling `GetSupportedMicroversions` all over the place.

Reply:
I only thing it will be more convenient to use methods on the wrapper rather than passing two strings around and parsing them each time. There is also a risk of people using the wrong order (what goes first, min or max)?

Reply:
Ah good point! I was still in my head thinking about how I did it in CAPO, where the min and max is stored in the ComputeClient and then these functions become methods on that. Maybe I can find some way to integrate this better with the ServiceClient :thinking: